mantidproject/mantid
# Changes suggested by modernize-use-auto and modernize-loop-convert

Description of work.

This addresses all remaining `modernize-use-auto` and `modernize-loop-convert` warnings from clang-tidy. Where possible and self-contained, I tried replacing raw loops with algorithms and new/delete with std::unique_ptr. 

**To test:**



Review changes and verify the builds pass.

This is a small change with no issue number.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
